### PR TITLE
Add stress-aware MLIP loss and stress-sign diagnostic

### PR DESCRIPTION
**Pull request by allaffa**
## Summary
- add an optional `stress_weight` term to energy-conserving MLIP training
- derive symmetric stress from the energy/cell-strain gradient and expose stress metrics during evaluation
- connect AllScAIP strain tracking to HydraGNN's MLIP wrapper
- add `diagnose_stress_sign` using six-component central finite differences
- document the stress data contract and sign-diagnostic workflow

## Validation
- `python3 -m compileall -q hydragnn tests/test_materials_preprocessing.py tests/test_interatomic_potential.py tests/_prediction_workflow.py`
- `git diff --check`
- Runtime pytest execution was not available in the local workspace because PyTorch and pytest are not installed.

## Configuration
```json
{
  "Architecture": {
    "enable_interatomic_potential": true,
    "energy_weight": 1.0,
    "energy_peratom_weight": 1.0,
    "force_weight": 10.0,
    "stress_weight": 1.0
  }
}
```

### CODE REVIEW

File: hydragnn/models/create.py [hunk 1]
```diff
@@ -971,8 +985,54 @@ def __getattr__(self, name):
...
+                    num_graphs = int(data.batch.max().item()) + 1
+                    cell = data.cell
+                    if cell.dim() == 2 and cell.shape == (3, 3):
+                        cell = cell.unsqueeze(0).expand(num_graphs, 3, 3).contiguous()
+                    elif cell.dim() == 2 and cell.shape[0] == 3 * num_graphs:
+                        cell = cell.view(num_graphs, 3, 3)
+                    elif cell.dim() != 3:
+                        raise ValueError(
+                            f"Unexpected cell shape {tuple(cell.shape)} for stress loss."
+                        )
```
[the remark this answers is not in the document]
Comment (allaffa) [reply hydragnn/models/create.py:1016]: Addressed in b3696f1.  cells are now expanded across the graph batch, and any other 3D batch-size mismatch raises a targeted . The stress-loss test now exercises the shared-cell case and the mismatch error.

File: hydragnn/train/train_validate_test.py [hunk 2]
```diff
@@ -1098,6 +1103,19 @@ def test(
...
+                            displacement = model.module._stress_displacement
+                            cell = model.module._stress_cell
+                            virial = torch.autograd.grad(
+                                graph_energy_pred,
+                                displacement,
+                                grad_outputs=torch.ones_like(graph_energy_pred),
+                                create_graph=False,
+                            )[0]
+                            stress_pred = virial / torch.det(cell).abs().view(-1, 1, 1)
+                            stress_true = data.stress.float().reshape_as(stress_pred)
```
[the remark this answers is not in the document]
Comment (allaffa) [reply hydragnn/train/train_validate_test.py]: Addressed in b3696f1. Evaluation now validates that volumes are finite and greater than dtype epsilon before division, matching the training-loss protection.

File: USER_MANUAL.md [hunk 3]
```diff
@@ -952,15 +952,18 @@ HydraGNN supports energy-conserving interatomic potential workflows. When enable
 ```json
 {
     "Architecture": {
-        "enable_interatomic_potential": true
+        "enable_interatomic_potential": true,
+        "energy_weight": 1.0,
+        "force_weight": 10.0,
+        "stress_weight": 1.0
     },
```
[the remark this answers is not in the document]
Comment (allaffa) [reply USER_MANUAL.md:960]: Addressed in b3696f1 by adding  to the user-manual example.

File: hydragnn/models/AllScAIPStack.py [hunk 4]
```diff
@@ -559,6 +559,10 @@ def _embedding(self, data):
 
         adapter = self._build_adapter(data)
         results = self.allscaip_backbone(adapter)
+        # Preserve the autograd strain handle and undeformed cell for the
+        # outer MLIP wrapper's optional stress loss.
+        data._hydragnn_stress_displacement = results["displacement"]
+        data._hydragnn_stress_cell = results["orig_cell"]
```
[the remark this answers is not in the document]
Comment (allaffa) [reply hydragnn/models/AllScAIPStack.py]: Addressed in b3696f1 by using results.get for both optional values. The current backbone always inserts these keys with None when disabled, so this was not presently a KeyError, but the change makes the adapter robust if that return contract changes.

File: hydragnn/models/create.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (allaffa) [reply hydragnn/models/create.py:1016]: Clarification: shared cells with shape (1, 3, 3) are expanded across the graph batch; other 3D batch-size mismatches raise a targeted ValueError.

File: USER_MANUAL.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (allaffa) [reply USER_MANUAL.md:960]: Clarification: energy_peratom_weight was added to the user-manual example in b3696f1.

File: hydragnn/models/create.py [hunk 5]
```diff
@@ -971,8 +985,62 @@ def __getattr__(self, name):
...
+                    if getattr(data, "cell", None) is None:
+                        raise ValueError(
+                            "stress_weight > 0 requires data.cell for every graph."
+                        )
+                    num_graphs = int(data.batch.max().item()) + 1
+                    cell = data.cell
+                    if cell.dim() == 2 and cell.shape == (3, 3):
+                        cell = cell.unsqueeze(0).expand(num_graphs, 3, 3).contiguous()
+                    elif cell.dim() == 2 and cell.shape[0] == 3 * num_graphs:
+                        cell = cell.view(num_graphs, 3, 3)
```
[the remark this answers is not in the document]
Comment (allaffa) [reply hydragnn/models/create.py]: Addressed in 8ad757b. The flattened batched-cell branch now requires the exact shape (3 * num_graphs, 3), and the test covers a malformed (3B, 2) input reaching the targeted ValueError.
